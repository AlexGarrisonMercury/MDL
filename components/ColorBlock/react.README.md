# Usage
Used to display a single section that displays a nicely spaced out text with a link inside a block of color and next to a large image.

<!-- PROPS -->
backgroundColor
colorScheme
copy
heading
image
link
linkText
subHeading