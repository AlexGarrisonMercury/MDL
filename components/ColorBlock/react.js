import React from 'react'
import classNames from 'classnames/bind'
import PropTypes from 'prop-types'
import styles from './style.module.scss'
import { color } from '@storybook/addon-knobs'

const cx = classNames.bind(styles)

const ColorBlock = ({
  link, linkText, heading, subHeading, copy, image, backgroundColor, colorScheme
}) => {
  return (
    <>
      <div className={cx('color-block')}>
        <figure className={cx('color-block--image')}>
          <a href={link} className={cx('color-block--image-link d-block')}>
            <img src={image} className={cx('color-block--image-img')} alt={'Color Example ' + heading} />
          </a>
        </figure>
        <div className={cx('color-block--text') + ' color-scheme-' + colorScheme} style={{backgroundColor: backgroundColor}}>
          <div className={cx('color-block--text-inner')}>
            <h3 className={cx('color-block--title')}>{heading}</h3>
            {subHeading && <h4 className={cx('color-block--subtitle')}>{subHeading}</h4>}
            <p className={cx('color-block--copy')}>{copy}</p>
            <a href={link} className={cx('color-block--link')}>{linkText}</a>
          </div>
        </div>
      </div>
    </>
  )
}

ColorBlock.defaultProps = {
  copy: null,
  heading: null,
  link: null,
  linkText: null,
  image: null,
  subHeading: null,
}

export default ColorBlock
