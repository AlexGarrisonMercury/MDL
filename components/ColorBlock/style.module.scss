@import 'styles/index';

.band-color-blocks {
  padding-top: 1rem;

  @media (min-width: 768px) {
    padding-top: 0;
  }
}

.color-block + .color-block {
  padding-top: 1rem;

  @media (min-width: 768px) {
    padding-top: 1.6rem;
  }

  @media (min-width: 1800px) {
    padding-top: 1.5rem;
  }
}

.color-block {
  @media (min-width: 768px) {
    display: flex;
    justify-content: space-between;
  }

  &:nth-child(even) {
    flex-direction: row-reverse;
  }
}

.color-block--image,
.color-block--text {
  @media (min-width: 768px) {
    width: 49%;
  }
}

.color-block--image {
  margin: 0 0 1rem;

  @media (min-width: 768px) {
    margin-bottom: 0;
    overflow: hidden;

    .color-block--image-img {
      height: 100%;
      max-width: none;
      width: auto;
    }
  }
}

.color-block--text {
  font-family: $font-family;
  padding: 2rem $space-l;

  @media (min-width: 768px) {
    display: flex;
    flex-direction: column;
    justify-content: center;
  }

  @media (min-width: 1281px) {
    padding-left: $space-xxl;
    padding-right: $space-xxl;
  }
}

.color-block--title {
  color: $color-m-black;
  font-size: 1.2rem;
  font-weight: 700;
  letter-spacing: 1.1px;
  margin-bottom: 1.2rem;
}

.color-block--copy {
  font-size: 14px;
  font-weight: 400;
  line-height: 22px;
  margin-bottom: 0.5rem;

  @media (min-width: 1281px) {
    font-size: 16px;
  }
}

.color-block--subtitle,
.color-block--copy {
  color: $color-m-black;
}

.color-block--title,
.color-block--link,
.color-block--copy {
  @media (min-width: 768px) {
    max-width: 100%;
  }
}

.color-block--link {
  color: $color-m-black;
  display: inline-block;
  font-size: 0.875rem;
  line-height: 1.5rem;
  margin-right: auto;
  text-decoration: none;
  position: relative;

  margin-bottom: 0.5rem;
  padding-top: 0.15rem;

  &:hover {
    color: $color-onyx;
    text-decoration: none;

    &::after {
      background: $color-limerick;
    }
  }

  &::after {
    position: absolute;
    content: '';
    height: 1px;
    bottom: 0px;
    left: 0;
    width: 100%;
    background: $color-slategray;
  }
}

.color-scheme-dark .color-block--subtitle,
.color-scheme-dark .color-block--copy,
.color-scheme-dark .color-block--link {
  color: $color-m-white;
}
