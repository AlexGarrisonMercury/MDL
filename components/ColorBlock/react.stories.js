import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, boolean, select } from '@storybook/addon-knobs';
import ColorBlock from './react';
import Readme from './react.README.md';

storiesOf('ColorBlock', module)
  .addParameters({ readme: { sidebar: Readme } })
  .addDecorator(withKnobs)
  .add('default', () => {
    return (
      <div style={{ margin: '0 auto', maxWidth: '1440px' }}>
        <ColorBlock
          link='#'
          linkText='Get Inspired'
          heading='More Life'
          subHeading=''
          copy={`We are constantly designing new and different ways for doors to do more. Whether it's more light, more peace and quiet, more peace of mind, more function, more technology or more style, we believe doors can do more.`}
          image='https://embed.widencdn.net/img/masonite/benaiziirh/708x429px/hero.jpeg'
          colorScheme='light'
          backgroundColor='#949fa6'
        />
        <ColorBlock
          link='#'
          linkText='Protect What Matters'
          heading='More Peace of Mind'
          subHeading=''
          copy={`Doors not only invite you in, they protect you once you're inside. And when it comes to our schools and universites, protection is paramount.`}
          image='https://embed.widencdn.net/img/masonite/j8aenj8pfa/708x429px/PeaceofMind_Img.jpg'
          colorScheme='light'
          backgroundColor='#92c83e'
        />
        <ColorBlock
          link='#'
          linkText='Explore MConnectÂ®'
          heading='More Connected'
          subHeading=''
          copy={`Our partnerships with customers, dealers, retailers, contractors and installers go way beyond doors. Join MConnectÂ® for tools, information, resources and solutions designed to create a smoother, faster, more friction-free experience.`}
          image='https://embed.widencdn.net/img/masonite/sp2z5baxpj/708x429px/More_Connected_Img.jpg'
          colorScheme='light'
          backgroundColor='#c7c1ab'
        />
      </div>
    );
  });
