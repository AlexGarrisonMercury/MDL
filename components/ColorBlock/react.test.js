import React from 'react'
import mount from 'enzyme'
import ColorBlock from './react'

describe('ColorBlock', () => {
  let wrapper

  beforeEach(() => {
    wrapper = mount(<ColorBlock />)
  })

  it('should render without crashing', () => {
    expect(asFragment()).toMatchSnapshot()
  })
})
