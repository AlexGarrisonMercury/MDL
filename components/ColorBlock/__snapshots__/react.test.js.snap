// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color Blocks should render without crashing 1`] = `
<DocumentFragment>
  <div
    class="color-block"
  >
    <figure
      class="color-block--image"
    >
      <a
        class="color-block--image-link d-block"
        href="#"
      >
        <img
          alt="Color Example More Life"
          class="color-block--image-img"
          src="https://embed.widencdn.net/img/masonite/benaiziirh/708x429px/hero.jpeg"
        />
      </a>
    </figure>
    <div
      class="color-block--text color-scheme-light"
      style="background-color: rgb(148, 159, 166);"
    >
      <div
        class="color-block--text-inner"
      >
        <h3
          class="color-block--title"
        >
          More Life
        </h3>
        <p
          class="color-block--copy"
        >
          We are constantly designing new and different ways for doors to do more. Whether it's more light, more peace and quiet, more peace of mind, more function, more technology or more style, we believe doors can do more.
        </p>
        <a
          class="color-block--link"
          href="#"
        >
          Get Inspired
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
